vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+analysis flag `ANL01FL`. Common options for `ANL01FL` would be to set
+null for invalid assessments or those occurring after new anti-cancer
+therapy, or to only flag assessments on or after after date of first
+treatment/randomization, or rules to cover the case when a patient has
+multiple observations per visit (e.g. by selecting worst value). Another
+consideration could be extra potential protocol-specific sources of
+Progressive Disease such as radiological assessments, which could be
+pre-processed here to create a PD record to feed downstream derivations.
+
+For the derivation of the parameters it is expected that the subject
+identifier variables (usually `STUDYID` and `USUBJID`) and `ADT` are a
+unique key. This can be achieved by deriving an analysis flag
+(`ANLzzFL`). See [Derive `ANL01FL`](#anl01fl) for an example.
+
+The below shows an example of a possible company-specific implementation
+of this step.
+
+### Select Overall Response Records and Set Parameter Details
+
+In this case we use the overall response records from `RS` from the
+investigator as our starting point. The parameter details such as
+`PARAMCD`, `PARAM` etc will always be company-specific, but an example
+is shown below so that you can trace through how these records feed into
+the other parameter derivations.
+
+```{r}
+adrs <- adrs %>%
+  filter(RSEVAL == "INVESTIGATOR" & RSTESTCD == "OVRLRESP") %>%
+  mutate(
+    PARAMCD = "OVR",
+    PARAM = "Overall Response by Investigator",
+    PARCAT1 = "Tumor Response",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "iRecist"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, VISIT, RSTESTCD, RSEVAL, PARAMCD, PARAM, PARCAT1, PARCAT2, PARCAT3)
+)
+```
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For
+this example, here we impute missing day to last possible date.
+
+```{r}
+adrs <- adrs %>%
+  derive_vars_dt(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    highest_imputation = "D",
+    date_imputation = "last"
+  ) %>%
+  mutate(AVISIT = VISIT)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, RSSTRESC, RSDTC, ADT, ADTF)
+)
+```
+
+### Derive `AVALC` and `AVAL`
+
+Here we populate `AVALC` and create the numeric version as `AVAL`
+(ordered from best to worst response). The `AVAL` values are not considered in
+the parameter derivations below, and so changing `AVAL` here would not change
+the result of those derivations.
+
+iRECIST ordering will be used or if you'd like to provide your own company-specific ordering here you
+could do this as follows:
+
+```{r}
+aval_resp_new <- function(arg) {
+  case_when(
+    arg == "iCR" ~ 8,
+    arg == "iPR" ~ 7,
+    arg == "iSD" ~ 6,
+    arg == "NON-iCR/NON-iUPD" ~ 5,
+    arg == "iCPD" ~ 4,
```

Comment:
As we are ordering from best to worst answer, iCPD should be after iUPD

Reply:
will update